controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java

```diff
@@ -728,6 +730,47 @@ private HttpResponse deleteSecretStore(String tenantName, String name, HttpReque
         return new SlimeJsonResponse(slime);
     }
 
+    private HttpResponse allowArchiveAccess(String tenantName, HttpRequest request) {
+        if (controller.tenants().require(TenantName.from(tenantName)).type() != Tenant.Type.cloud)
+            throw new IllegalArgumentException("Tenant '" + tenantName + "' is not a cloud tenant");
+
+        var data = toSlime(request.getData()).get();
+        var role = mandatory("role", data).asString();
+
+        var tenant = (CloudTenant) controller.tenants().require(TenantName.from(tenantName));
```

Comment:
Looks like we are just using this to get the `TenantName`, but we already have it?